Dockerfile

```diff
@@ -10,15 +10,38 @@ USER node
 
 RUN yarn install
 
-RUN yarn bootstrap
+RUN yarn workspace @zooniverse/react-components build
 
-FROM node:12 AS production-apps
+RUN yarn workspace @zooniverse/classifier build
 
-RUN mkdir -p /usr/src
-WORKDIR /usr/src/
+ARG COMMIT_ID
+ENV COMMIT_ID=$COMMIT_ID
 
-COPY --from=bootstrap /usr/src /usr/src
+ENV PANOPTES_ENV production
 
-RUN yarn workspace @zooniverse/fe-project build
+ENV NODE_ENV production
```

Comment:
Does anyone know why `NODE_ENV` is explicitly set for the app builds?
https://github.com/zooniverse/front-end-monorepo/blob/6b73ad555eb98799283f28950e4517d74d029b57/packages/app-project/Dockerfile#L15-L17

Reply:
Previously, this was set in the Jenkins build environment (see #1494) but `yarn install` ignores dev dependencies during a production build, so that setup would break the library builds (there'd be no webpack installed, for a start.)
https://classic.yarnpkg.com/en/docs/cli/install/#toc-yarn-install-production-true-false

Setting `NODE_ENV` for the image in the Dockerfile builds production builds of the NextJS apps, but clobbers any tests that use `NODE_ENV=test`. You can try that out on this branch with `docker-compose run --rm bash`, then run the tests in the built image with `yarn test`. That might not be a big deal for us, because we don't run our CI tests in the production Docker images.

Reply:
Maybe `NODE_ENV` should be a build argument?

Reply:
Do you remember why you hardcoded them in #1494? I think it should be okay to set them via injected args, but we'd have to add  `--production=false` to the install step.

Reply:
hard coded is the default without the injected args but allows for them to be overriden at build time, https://github.com/zooniverse/front-end-monorepo/pull/1494/files#r377658763